sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
+    private readonly institutionLocationService: InstitutionLocationService,
+    private readonly confirmationOfEnrollmentService: ConfirmationOfEnrollmentService,
+    private readonly systemUsersService: SystemUsersService,
   ) {
     this.institutionIntegrationConfig = config.institutionIntegration;
   }
 
   /**
    * Process all the available ECE response files in SFTP location.
    * Once the file is processed, it gets deleted.
-   * @returns Process summary.
+   * @returns Process summary result.
    */
   async process(): Promise<ProcessSummaryResult[]> {
-    // Get the list of all files from SFTP ordered by file name.
-    const fileSearch = new RegExp(
-      `^\\/\\w{4}\\/${ECE_RESPONSE_FILE_NAME}$`,
-      "i",
+    // Get all the institution codes who are enabled for integration.
+    const integrationEnabledInstitutions =
+      await this.institutionLocationService.getAllIntegrationEnabledInstitutionCodes();
+    const filePaths = integrationEnabledInstitutions.map((institutionCode) =>
+      path.join(
+        this.institutionIntegrationConfig.ftpResponseFolder,
+        institutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      ),
     );
-    const filePaths = await this.integrationService.getResponseFilesFullPath(
-      this.institutionIntegrationConfig.ftpResponseFolder,
-      fileSearch,
+    // Process all the files in parallel.
+    const result = await processInParallel<ProcessSummaryResult, string>(
+      (remoteFilePath: string) =>
+        this.processDisbursementsInECEResponseFile(remoteFilePath),
+      filePaths,
     );
-    const result: ProcessSummaryResult[] = [];
-    // TODO: Processing of ECE files to be implemented.
-    // This code will be updated.
-    for (const filePath of filePaths) {
-      result.push({
-        summary: [`Reading file ${filePath}`],
-        warnings: [],
-        errors: [],
-      });
-    }
     return result;
   }
 
+  /**
+   * Process individual ECE response file sent by institution.
+   * @param remoteFilePath ECE response file path.
+   * @returns process summary result.
+   */
+  private async processDisbursementsInECEResponseFile(
+    remoteFilePath: string,
+  ): Promise<ProcessSummaryResult> {
+    const processSummary: ProcessSummaryResult = new ProcessSummaryResult();
+    let isECEResponseFileExist = false;
+    processSummary.summary.push(`Starting download of file ${remoteFilePath}.`);
+    this.logger.log(`Starting download of file ${remoteFilePath}.`);
+    try {
+      const eceFileDetailRecords =
+        await this.integrationService.downloadResponseFile(remoteFilePath);
+      isECEResponseFileExist = !!eceFileDetailRecords.length;
+      // Check if the file exist in remote server and summary info
+      // if the file is not found.
+      if (!isECEResponseFileExist) {
+        const warningMessage = `File ${remoteFilePath} not found.`;
+        processSummary.summary.push(warningMessage);
```

Comment:
When the `QueueProcessSummary` was created the idea was to avoid doing this type of log when we need white the log to both places. In the previous PR the `ProcessSummaryResult` was centralized, which was already a win, but having the `QueueProcessSummary` also centralized was the final goal to have these logs simplifies and no longer duplicated.
If possible, let's have a chat about to possibility handle this now or in the near future.
@ann-aot @guru-aot @andrepestana-aot @sh16011993 FYI.